server/src/main/java/io/spine/server/integration/ThirdPartyContext.java

```diff
@@ -0,0 +1,190 @@
+     *
+     * @param name
+     *         name of the third-party system
+     */
+    public static ThirdPartyContext multitenant(String name) {
+        checkNotNull(name);
+        return newContext(name, true);
+    }
+
+    private static ThirdPartyContext newContext(String name, boolean multitenant) {
+        BoundedContext context = notStoringEvents(name, multitenant).build();
+        context.integrationBus()
+               .notifyOfCurrentNeeds();
+        return new ThirdPartyContext(context);
+    }
+
+    private ThirdPartyContext(BoundedContext context) {
+        this.context = context;
+        this.producerId = pack(context.name());
+    }
+
+    /**
+     * Emits an event from the third-party system.
+     *
+     * <p>If the event is required by another Context, posts the event into
+     * the {@link IntegrationBus} of the respective Context. Does nothing if the event is not
+     * required by any Context.
+     *
+     * <p>The caller is required to supply the tenant ID via the {@code ActorContext.tenant_id} if
+     * this Context is multitenant.
+     *
+     * @param actorContext
+     *         the info about the actor, a user or a software component, who emits the event
+     * @param eventMessage
+     *         the event
+     */
+    public void emittedEvent(ActorContext actorContext, EventMessage eventMessage) {
+        checkNotNull(actorContext);
+        checkNotNull(eventMessage);
+        checkTenant(actorContext, eventMessage);
+
+        EventId id = EventId
+                .newBuilder()
+                .setValue(newUuid())
+                .build();
+        EventContext eventContext = EventContext
+                .newBuilder()
+                .setProducerId(producerId)
+                .setTimestamp(actorContext.getTimestamp())
+                .setImportContext(actorContext)
+                .setExternal(true)
+                .vBuild();
+        Event event = Event
+                .newBuilder()
+                .setId(id)
+                .setContext(eventContext)
+                .setMessage(pack(eventMessage))
+                .vBuild();
+        context.eventBus()
+               .post(event);
+    }
+
+    /**
+     * Emits an event from the third-party system.
+     *
+     * <p>If the event is required by another Context, posts the event into
+     * the {@link IntegrationBus} of the respective Context. Does nothing if the event is not
+     * required by any Context.
+     *
+     * <p>This overload may only be used for single-tenant third-party contexts. If this Context is
+     * multitenant, this method throws an exception.
+     *
+     * @param userId
+     *         the ID of the actor, a user or a software component, who emits the event
+     * @param eventMessage
+     *         the event
+     */
+    public void emittedEvent(UserId userId, EventMessage eventMessage) {
+        checkNotNull(userId);
```

Comment:
Please consider for checking for non-default too.

Reply:
It is checked as a part of `ActorContext` validation.